pkg/snapstore/snapstore_test.go

```diff
@@ -278,6 +280,220 @@ var _ = Describe("Save, List, Fetch, Delete from mock snapstore", func() {
+					newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+					Expect(err).Should(HaveOccurred())
+					Expect(newSecretModifiedTime.IsZero()).Should(BeTrue())
+				})
+			})
+			Context("environment variable set", func() {
+				var credentialDirectory string
+				BeforeEach(func() {
+					credentialDirectory = GinkgoT().TempDir()
+					GinkgoT().Setenv(config.EnvVariable, credentialDirectory)
+					// config.CredentialType == "file" -> env variable set to file
+					if config.CredentialType == "file" {
+						GinkgoT().Setenv(config.EnvVariable, filepath.Join(credentialDirectory, config.CredentialFiles[0]))
+					}
+				})
+				Context("credentials do not exist", func() {
+					// all files are missing
+					It("should return error when all files are missing", func() {
+						newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+						Expect(err).Should(HaveOccurred())
+						Expect(newSecretModifiedTime.IsZero()).Should(BeTrue())
+					})
+					// one file is missing
+					for _, credentialFile := range config.CredentialFiles {
+						credentialFile := credentialFile
+						It(fmt.Sprintf("should return error when the file '%s' is missing", credentialFile), func() {
+							// create all credential files first
+							lastCreationTime, err := createCredentialFilesInDirectory(credentialDirectory, config.CredentialFiles)
+							Expect(err).ShouldNot(HaveOccurred())
+							Expect(lastCreationTime.IsZero()).ShouldNot(BeTrue())
+							err = os.Remove(filepath.Join(credentialDirectory, credentialFile))
+							Expect(err).ShouldNot(HaveOccurred())
+							// remove one credential file and then run the test
+							newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+							Expect(err).Should(HaveOccurred())
+							Expect(newSecretModifiedTime.IsZero()).Should(BeTrue())
+						})
+					}
+				})
+				Context("credentials exist", func() {
+					var lastCreationTime time.Time
+					BeforeEach(func() {
+						var err error
+						lastCreationTime, err = createCredentialFilesInDirectory(credentialDirectory, config.CredentialFiles)
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(lastCreationTime.IsZero()).ShouldNot(BeTrue())
+					})
+					// unmodified credentials
+					It("should return the latest creation time among the credential files", func() {
+						newSecretModifiedTime, err := GetSnapstoreSecretModifiedTime(config.SnapstoreProvider)
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(newSecretModifiedTime.Equal(lastCreationTime)).Should(BeTrue())
+					})
+					// modified credentials
+					for _, credentialFile := range config.CredentialFiles {
+						credentialFile := credentialFile
+						It(fmt.Sprintf("should return the modification time of the credential file %s", credentialFile), func() {
+							fmt.Println("Testing dynamic credential rotation for the provider:", config.Provider)
```

Comment:
```suggestion
						It(fmt.Sprintf("should return the modification time of the credential file %q", credentialFile), func() {

```

Reply:
Addressed in [8635c10](https://github.com/gardener/etcd-backup-restore/pull/670/commits/8635c101d031f890343da3177cb410aa20053dc5)